### PR TITLE
3506 - 0

### PR DESCRIPTION
**Pull request by als6k**
## Номер
3506

## Номер задания
0

## Ссылка на видео с демо
https://youtu.be/Bf3idqWvcHs

### CODE REVIEW

File: 3506/0/ruby_koans/about_scoring_project.rb [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+#
+# More scoring examples are given in the tests below:
+#
+# Your goal is to write the score method.
+
+def score(dice)
+  # [1,1,1,3,5]
+  result = 0
+
+  nums = dice.inject({}) do |h,n|
```
Comment (DmitryAvramec) [inline 3506/0/ruby_koans/about_scoring_project.rb]: `each_with_object`?

File: 3506/0/ruby_koans/about_scoring_project.rb [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+# score([1,1,1,5,1]) => 1150 points
+# score([2,3,4,6,2]) => 0 points
+# score([3,4,5,3,3]) => 350 points
+# score([1,5,1,2,4]) => 250 points
+#
+# More scoring examples are given in the tests below:
+#
+# Your goal is to write the score method.
+
+def score(dice)
```
Comment (DmitryAvramec) [inline 3506/0/ruby_koans/about_scoring_project.rb]: Style!
This is simple math! You don't need so many actions for processing this one.
Rewrite it

File: 3506/0/ruby_koans/triangle.rb [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+# and
+#   about_triangle_project_2.rb
+#
+def triangle(a, b, c)
+  # WRITE THIS CODE
+  a, b, c = [a, b, c].sort
+  raise TriangleError if a <= 0
+  raise TriangleError if a + b <= c
+
+  return :equilateral if a == b && b == c
```
Comment (DmitryAvramec) [inline 3506/0/ruby_koans/triangle.rb]: Why we need these conditions if we already have sorted array? Use Array methods for moving out from complex conditions

File: 3506/0/ruby_koans/triangle.rb [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+# and
+#   about_triangle_project_2.rb
+#
+def triangle(side_a, side_b, side_c)
+  # WRITE THIS CODE
+  side_a, side_b, side_c = [side_a, side_b, side_c].sort
+  raise TriangleError if side_a <= 0
+  raise TriangleError if side_a + side_b <= side_c
+
+  return :equilateral if side_a == side_b
```
Comment (DmitryAvramec) [inline 3506/0/ruby_koans/triangle.rb]: Use array

File: 3506/0/ruby_koans/about_dice_project.rb [hunk 5]
```diff
@@ -0,0 +1,66 @@
+require File.expand_path(File.dirname(__FILE__) + '/neo')
+
+# Implement a DiceSet Class here:
+#
+class DiceSet
+  attr_reader :values
+  def roll(num)
+    @values = (1..num).map { rand(1..6) }
```
Comment (DmitryAvramec) [inline 3506/0/ruby_koans/about_dice_project.rb]: Array.new?

File: 3506/0/ruby_koans/about_scoring_project.rb [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+  # Example: [1,1,1,3,5]
+  result = 0
+  nums = find_nums(dice)
+  set3 = nums.find { |_k, v| v >= 3 } # [1,3]
```
Comment (DmitryAvramec) [inline 3506/0/ruby_koans/about_scoring_project.rb]: You dont need this find, in ruby you have mote simple ways

File: 3506/0/ruby_koans/triangle.rb (hunk 4 above)
Comment (als6k) [reply 3506/0/ruby_koans/triangle.rb]: Can you elaborate on that?

Comment (DmitryAvramec) [reply 3506/0/ruby_koans/triangle.rb]: Think about using` .uniq `method

File: 3506/0/ruby_koans/triangle.rb [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+#
+# :reek:TooManyStatements
+def triangle(side_a, side_b, side_c)
+  side_a, side_b, side_c = [side_a, side_b, side_c].sort
+  raise TriangleError if side_a <= 0
+  raise TriangleError if side_a + side_b <= side_c
+
+  array = [side_a, side_b, side_c].uniq
+  return :equilateral if array.size == 1
+  return :isosceles if array.size == 2
```
Comment (DmitryAvramec) [inline 3506/0/ruby_koans/triangle.rb]: By the way, you can make it without return) Just think about merging this 4 rows together

File: 3506/0/ruby_koans/about_scoring_project.rb [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+    if count >= 3
+      result = (num == 1 ? 1000 : num * 100)
+      count -= 3
+    end
+    result = count_single_nums(result, num, count)
+  end
+  result
+end
+
+def count_single_nums(result, num, count)
```
Comment (DmitryAvramec) [inline 3506/0/ruby_koans/about_scoring_project.rb]: Tell me about the sacred meaning of this method...

File: 3506/0/ruby_koans/about_scoring_project.rb [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+  result = 0
+  nums = Hash.new(0)
+  dice.each do |i|
+    nums[i] = (nums[i] || 0) + 1
+  end
+  result = count_points(nums, result)
+  result
+end
+
+def count_points(nums, result)
```
Comment (DmitryAvramec) [inline 3506/0/ruby_koans/about_scoring_project.rb]: I try to guide you. But you going to the dark side
Let's combine all 3 methods together:
```
 def score (dice)
  (1..6).each_with_object(0) do |number, score|
    # dice.count(number):
    # [1,1,1,1,2].count(1) => 4
    # TODO: Implement other code bellow
  end
end
```

File: 3506/0/ruby_koans/about_scoring_project.rb [hunk 10]
```diff
@@ -29,33 +29,24 @@
 #
 # Your goal is to write the score method.
 
+# rubocop:disable Metrics/MethodLength
```
Comment (DmitryAvramec) [inline 3506/0/ruby_koans/about_scoring_project.rb]: Remove this disable

File: 3506/0/ruby_koans/about_scoring_project.rb [hunk 11]
```diff
@@ -29,33 +29,24 @@
...
 #:reek:all:
 def score(dice)
   result = 0
```
Comment (DmitryAvramec) [inline 3506/0/ruby_koans/about_scoring_project.rb]: You dont need this var cos you use `each_with_object(0)`

**Pull request merged**